app/assets/javascripts/modules/remove-filter.js

```diff
@@ -33,7 +33,18 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
         $input.trigger('change');
       }
       else if (inputType == 'text' || inputType == 'search') {
-        $input.val(currentVal.replace(removeFilterValue, '').replace(/\s+/g,' ').trim()).trigger({
+        var splitKeywords = currentVal.split(" ");
+
+        for (var i = 0; splitKeywords.length; i += 1) {
+          if (splitKeywords[i].toString() === removeFilterValue.toString()) {
+            splitKeywords.splice(i, 1);
+            break;
+          }
+        }
+
+        var newVal = splitKeywords.join(" ")
+
+        $input.val(newVal.trim()).trigger({
```

Comment:
Nitpick: I'd add the `trim` onto the end of line 45, just to make this line only about setting a value.